File: Sources/ClaudeUsageWidgetCore/Usage/ThresholdLevel.swift
/// How alarming a utilization fraction is. Kept separate from the palette so
/// the thresholds are testable without touching SwiftUI.
public enum ThresholdLevel: Equatable, Sendable {
    case ok
    case warning
    case danger

    public static func level(for fraction: Double) -> ThresholdLevel {
        switch fraction {
        case ..<0.6: .ok
        case ..<0.85: .warning
        default: .danger
        }
    }
}

File: src/ClaudeUsageWidget.Core/Usage/ThresholdLevel.cs
namespace ClaudeUsageWidget.Core;

/// How alarming the usage fraction is. Kept separate from the color palette
/// so the thresholds can be tested without touching the UI.
public enum ThresholdLevel
{
    Ok,
    Warning,
    Danger,
}

public static class Thresholds
{
    public static ThresholdLevel Level(double fraction) =>
        fraction < 0.6 ? ThresholdLevel.Ok
        : fraction < 0.85 ? ThresholdLevel.Warning
        : ThresholdLevel.Danger;
}
